sentence_transformers/sparse_encoder/SparseEncoder.py

```diff
@@ -1058,39 +1058,107 @@ def intersection(
+            values = embeddings.values()
+            indices = embeddings.indices().squeeze()
+            if values.numel() == 0:
+                return []
+
+            # Apply top-k if specified
+            if top_k is not None:
+                top_values, top_idx = torch.topk(values, min(top_k, values.numel()))
+                indices = indices[top_idx]
+                values = top_values
+
+            # Convert token IDs to strings
+            tokens = self.tokenizer.convert_ids_to_tokens(indices.tolist())
+
+            # Return a dictionary mapping tokens to weights
+            return list(zip(tokens, values.tolist()))
+
+        # For a batch of embeddings
+        elif embeddings.dim() == 2:
+            embeddings = embeddings.coalesce() if embeddings.is_sparse else embeddings
+
+            # Extract indices and values
+            indices = embeddings.indices()
+            values = embeddings.values()
+
+            if values.numel() == 0:
+                return [{}] * embeddings.size(0)
+
+            # Sample indices (first dimension) and token indices (second dimension)
+            sample_indices, token_indices = indices[0], indices[1]
+
+            # Count tokens per sample
+            sample_counts = torch.bincount(sample_indices, minlength=embeddings.size(0)).tolist()
+
+            # Apply top-k if specified
+            if top_k is not None:
+                results = []
+                start_idx = 0
+                for i, count in enumerate(sample_counts):
+                    if count == 0:
+                        results.append({})
+                        continue
+
+                    sample_values = values[start_idx : start_idx + count]
+                    sample_tokens = token_indices[start_idx : start_idx + count]
+
+                    if count > top_k:
+                        top_values, top_idx = torch.topk(sample_values, top_k)
+                        top_tokens = sample_tokens[top_idx]
+                        token_strs = self.tokenizer.convert_ids_to_tokens(top_tokens.tolist())
+                        results.append(list(zip(token_strs, top_values.tolist())))
+                    else:
+                        token_strs = self.tokenizer.convert_ids_to_tokens(sample_tokens.tolist())
+                        results.append(list(zip(token_strs, sample_values.tolist())))
+
+                    start_idx += count
+
+                return results
             else:
-                top_values, top_indices = torch.topk(embeddings, top_k)
-                top_tokens = tokenizer.convert_ids_to_tokens(top_indices.tolist())
-                return list(zip(top_tokens, top_values.tolist()))
+                # Process all tokens for each sample
+                results = []
+                start_idx = 0
+                for i, count in enumerate(sample_counts):
+                    if count == 0:
+                        results.append({})
```

Comment:
```suggestion
                        results.append([])
```
Again, a dict doesn't really make sense